modules/library/coverage/src/main/java/org/geotools/image/util/ImageUtilities.java

```diff
@@ -1303,10 +1319,30 @@ public static void disposeSinglePlanarImage(PlanarImage planarImage) {
      */
     private static void cleanField(Object theObject, String fieldName)
             throws NoSuchFieldException, IllegalAccessException {
-        Field f = theObject.getClass().getDeclaredField(fieldName);
+        cleanField(theObject, fieldName, false);
+    }
+
+    /**
+     * Helper that cleans up a field
+     *
+     * @param theObject
+     * @param fieldName
+     * @throws NoSuchFieldException
+     * @throws IllegalAccessException
+     */
+    private static void cleanField(Object theObject, String fieldName, boolean superClass)
+            throws NoSuchFieldException, IllegalAccessException {
+        // getDeclaredField only provides access to current class,
+        // not superclass.
+        Class<? extends Object> theClass = theObject.getClass();
```

Comment:
Why not use the getField method instead? It goes up the inheritance chain by itself, no need for flags

Reply:
getField only get public fields, not private ones :(